doc/create-cloud-resources.rst

```diff
@@ -0,0 +1,145 @@
+     more information on the gcloud SDK.
+   - Install ``kubectl``, which is a tool for controlling kubernetes. From
+     the terminal, enter:
+
+     .. code-block:: bash
+
+        gcloud components install kubectl
+
+4. Create a Kubernetes cluster on Google Cloud, by typing in the following
+   command:
+
+   .. code-block:: bash
+
+      gcloud container clusters create <YOUR_CLUSTER> \
+          --num-nodes=3 \
+          --machine-type=n1-standard-2 \
+          --zone=us-central1-b
+
+   where:
+
+   * ``--num-nodes`` specifies how many computers to spin up. The higher the
+     number, the greater the cost.
+   * ``--machine-type`` specifies the amount of CPU and RAM in each node. There
+     is a `variety of types <https://cloud.google.com/compute/docs/machine-types>`_
+     to choose from. Picking something appropriate here will have a large effect
+     on how much you pay - smaller machines restrict the max amount of RAM each
+     user can have access to but allow more fine-grained scaling, reducing cost.
+     The default (`n1-standard-2`) has 2CPUs and 7.5G of RAM each, and might not
+     be a good fit for all use cases!
+   * ``--zone`` specifies which data center to use. Pick something that is not
+     too far away from your users. You can find a list of them `here <https://cloud.google.com/compute/docs/regions-zones/regions-zones#available>`_.
+
+5. To test if your cluster is initialized, run:
+
+   .. code-block:: bash
+
+      kubectl get node
+
+   The response should list three running nodes.
+
+Next we'll install a few tools that are required for BinderHub to run properly.
+
+Installing Helm
+---------------
+
+Next, we'll install **Helm**. This allows us to control our Kubernetes cluster
+with a configuration file (called a Helm Chart). By using a **Helm Chart**, we
+can set up the cluster deployment to have the resources necessary for
+running BinderHub.
+
+Run the following commands to download and install helm::
+
+   curl https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get | bash
+   helm init
+
+Install kubectl
+---------------
+
+Next we'll install ``kubectl`` (short for Kubernetes Control). This allows us
+to interact with the Kubernetes instance, and to get information about what
+nodes are running on our Kubernetes platform. Run the following command::
+
+   gcloud components install kubectl
+
+.. note::
+
+   If you're working from within Google Cloud Console, this will already be
+   installed.
+
+.. _setup-registry:
+
+Set up the container registry
+-----------------------------
+
+BinderHub will build Docker images out of GitHub repositories, and then
+register those images with an online registry so that JupyterHub can
+serve user instances from that registry. You can use any registry that
+you like, though this guide covers how to properly configure the **Google
+Container Registry** (``gcr.io``).
+
+Doing this involves using the Container Registry user interface in google
```

Comment:
You need to provide BinderHub with proper credentials so it can push images to the Google Container Registry. You can do so by creating a **service account** that has authorization to push to Google Container Registry: